### PR TITLE
enh(Listing): Use CSS grid instead of html table in Listing

### PR DESCRIPTION
**Pull request by bdauria**
This improves the responsiveness and makes the column definition a lot more flexible (possibility to use css grid width attributes like 20px, auto, 100% or 1fr)

### CODE REVIEW

File: src/Listing/index.tsx [hunk 1]
```diff
@@ -187,7 +210,27 @@ const Listing = ({
...
-    return `calc(100vh - ${tableTopOffset}px - ${paginationElement.current?.clientHeight}px - ${tableHeaderElement.current?.clientHeight}px)`;
+    return `calc(100vh - ${tableTopOffset}px - ${
+      actionBarRef.current?.offsetHeight
+    }px - ${headerHeight}px - ${loadingIndicatorHeight}px - ${theme.spacing(
+      1,
+    )}px)`;
+  };
+
+  const getGridTemplateColumn = (): string => {
+    const checkbox = checkable ? 'auto ' : '';
```
Comment (Thebarda) [inline src/Listing/index.tsx:221]:
```suggestion
    const checkbox = checkable ? 'min-content' : '';
```
That avoid the checkbox to take a lot of space

Comment (kduret) [reply src/Listing/index.tsx:221]: don't forget space after `min-content`

**Review by Thebarda — APPROVED** [review]

File: src/Listing/Cell/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      if (isRowHovered) {
+        return fade(theme.palette.primary.main, 0.08);
+      }
+
+      const foundCondition = rowColorConditions?.find(({ condition }) =>
+        condition(row),
+      );
+
+      if (!isNil(foundCondition)) {
+        return foundCondition.color;
```
Comment (Thebarda) [inline src/Listing/Cell/index.tsx:22]:
```suggestion
        return foundCondition?.color;
```

Comment (bdauria) [reply src/Listing/Cell/index.tsx:22]: The nil check is already done above

File: src/Listing/Cell/DataCell.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  };
+
+  return cellByColumnType[column.type]();
+};
+
+const MemoizedDataCell = React.memo<Props>(DataCell, (prevProps, nextProps) => {
+  const previousColumn = prevProps.column;
+  const previousRow = prevProps.row;
+  const previousIsRowHovered = prevProps.isRowHovered;
+  const previousIsRowSelected = prevProps.isRowSelected;
```
Comment (kduret) [inline src/Listing/Cell/DataCell.tsx:122]: we don't have rule for this now, but I think it's more readable like this : 
```suggestion
  const {
    column: previousColumn,
    row: previousRow,
    isRowHovered: previousIsRowHovered,
    isRowSelected: previousIsRowSelected,
  } = prevProps;
```

File: src/Listing/Cell/DataCell.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  }
+
+  return (
+    equals(previousIsComponentHovered, nextIsComponentHovered) &&
+    equals(previousIsRowHovered, nextIsRowHovered) &&
+    equals(previousFormattedString, nextFormatttedString) &&
+    equals(previousColSpan, nextColSpan) &&
+    equals(previousIsTruncated, nextisTruncated) &&
+    equals(previousHiddenCondition, nextHiddenCondition) &&
+    equals(previousHiddenCondition, nextHiddenCondition) &&
```
Comment (kduret) [inline src/Listing/Cell/DataCell.tsx:183]: duplicated line

File: src/Listing/Cell/DataCell.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
```
Comment (bdauria) [inline src/Listing/Cell/DataCell.tsx:185]:
```suggestion
```

File: src/Listing/Cell/DataCell.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  ) {
+    return true;
+  }
+
+  return (
+    equals(previousIsComponentHovered, nextIsComponentHovered) &&
+    equals(previousIsRowHovered, nextIsRowHovered) &&
+    equals(previousFormattedString, nextFormatttedString) &&
+    equals(previousColSpan, nextColSpan) &&
+    equals(previousIsTruncated, nextisTruncated) &&
```
Comment (kduret) [inline src/Listing/Cell/DataCell.tsx:183]:
```suggestion
    equals(previousIsTruncated, nextIsTruncated) &&
```

File: src/Listing/Cell/DataCell.tsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+  const previousIsTruncated = previousColumn.isTruncated;
+  const previousColSpan = previousColumn.getColSpan?.(previousIsRowSelected);
+  const previousHiddenCondition = previousColumn.getHiddenCondition?.(
+    previousIsRowSelected,
+  );
+
+  const nextColumn = nextProps.column;
+  const nextRow = nextProps.row;
+  const nextIsRowHovered = nextProps.isRowHovered;
+  const nextIsRowSelected = nextProps.isRowSelected;
```
Comment (kduret) [inline src/Listing/Cell/DataCell.tsx:146]:
```suggestion
  const {
    column: nextColumn,
    row: nextRow,
    isRowHovered: nextIsRowHovered,
    isRowSelected: nextIsRowSelected,
  } = nextProps;
```

File: src/Listing/Cell/DataCell.tsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+  const nextIsComponentHovered = nextHasHoverableComponent && nextIsRowHovered;
+
+  const nextFormatttedString = nextColumn.getFormattedString?.(nextRow);
+
+  const nextColSpan = nextColumn.getColSpan?.(nextIsRowSelected);
+
+  const nextHiddenCondition = nextColumn.getHiddenCondition?.(
+    nextIsRowSelected,
+  );
+  const nextisTruncated = nextColumn.isTruncated;
```
Comment (kduret) [inline src/Listing/Cell/DataCell.tsx:163]:
```suggestion
  const nextIsTruncated = nextColumn.isTruncated;
```

**Review by kduret — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**